rust/lance/src/index/vector.rs

```diff
@@ -292,6 +293,17 @@ pub(crate) async fn remap_vector_index(
     Ok(())
 }
 
+#[instrument(level = "debug", skip_all, fields(new_uuid = new_uuid.to_string(), dataset_uri = dataset.base.to_string(), column = from_field.name))]
+pub(crate) async fn cast_vector_index(
+    dataset: &Dataset,
+    index: Arc<dyn Index>,
+    new_uuid: &Uuid,
+    from_field: &Field,
+    to_field: &Field,
+) -> Result<()> {
+    todo!()
+}
```

Comment:
@eddyxu What's your advice on the easiest way to implement this? It probably just needs to work on IVF_PQ indices for now. This is for casting f32 index to f16.

Reply:
Can we hold a `Arc<dyn Ivf>`? Use match to downcast Index to known types might work.

Reply:
What I mean is, I want to write basically:

```rust
{
    let old_index = dataset
        .open_vector_index(&from_field.name, &old_uuid.to_string())
        .await?;

    let new_index = old_index.cast(&to_field.into())?;

    new_index
        .save(dataset, &new_uuid.to_string(), to_field)
        .await?;
    
    Ok(())
}
```

But there is no `save()` method on an index or anything like it. How can I mutate an index?